### PR TITLE
[zh-cn]: update the translation of "Building an Internet-Connected Phone with PeerJS"

### PR DESCRIPTION
**Pull request by tianyeeT**
### Related issues and pull requests
#16473

### CODE REVIEW

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   - `call.answer(window.localStream)`：如果 `answerCall` 是 `true`，你将想要调用 peerJS 的 `answer()` 函数来接听通话，传递本地流作为参数。
+   - `showCallContent`：类似于你在呼叫按钮事件监听器中所做的，你希望被呼叫者看到正确的 HTML 内容。
+   - `call.on('stream', () => { })` 块中的所有内容与呼叫按钮的事件监听器中的内容完全相同。之所以在这里也要添加它，是为了让浏览器也更新被呼叫者的视图。
+   - 如果被呼叫者拒绝了通话，我们只会在控制台中记录一条消息。
+
+3. 现在你已经有足够的代码来创建通话并接听它。刷新你的浏览器并进行测试。确保两个浏览器都打开了控制台，否则你将无法获得回答通话的提示。点击呼叫，提交另一个浏览器的对等 ID，然后接听通话。最终页面应该如下所示：
+
+   ![两个屏幕并排，都是深奶油色的背景，用深绿色的粗体字显示着“给朋友打电话”。紧接着是“你已连接”的字样，再下方是“请使用耳机！”和“你已自动静音，请取消静音！”。接着是一个大的深红色按钮，上面用与背景相同的深奶油色写着“挂断”。](screens_side_by_side.png)
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Creating_a_call", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/End_a_call")}}
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md:60]: 文档的末尾需要有个空行

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     proxied: true,
+     debug: true,
+     path: "/myapp",
+     ssl: {},
+   });
+   ```
+
+   我们使用 `ExpressPeerServer` 对象来创建 Peer 服务器，在此过程中传递一些选项。Peer 服务器将处理我们所需的 WebRTC 信令，因此我们无需担心 STUN/TURN 服务器或其他协议。
+
+4. 最后，你需要通过调用 `app.use(peerServer)` 告诉你的应用程序使用 `peerServer`。你完成的 `server.js` 应包括在服务器文件中包含的其他必要依赖项，以及将 `index.html` 文件提供给根路径。
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.md:34]:
```suggestion
4. 最后，你需要通过调用 `app.use(peerServer)` 告诉你的应用程序使用 `peerServer`。你完成的 `server.js` 应包括在服务器文件中包含的其他必要依赖项，以及将 `index.html` 文件提供给根路径。

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+   ```js
+   peer.on("call", (call) => {
+     const answerCall = confirm("Do you want to answer?");
+   });
+   ```
+
+   首先，我们使用确认提示（confirm prompt）提示用户进行回答。这将在屏幕上显示一个窗口（如图所示），用户可以选择“确定”或“取消” — 这对应于返回的布尔值。当你在浏览器中点击“呼叫”时，将出现以下提示：
+
+   ![A browser prompt that asks "Do you want to answer?" with two options: "Cancel" and "Ok"](confirm_prompt.png)
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md:24]:
```suggestion
   ![一个浏览器提示框，内容为“你想回答吗？”，有两个选项：“取消”和“确定”。](confirm_prompt.png)
```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+1. PeerJS 框架使 `.on('call')` 事件可用，让我们在这里使用它。将此添加到 `script.js` 文件的底部：
+
+   ```js
+   peer.on("call", (call) => {
+     const answerCall = confirm("Do you want to answer?");
+   });
+   ```
+
+   首先，我们使用确认提示（confirm prompt）提示用户进行回答。这将在屏幕上显示一个窗口（如图所示），用户可以选择“确定”或“取消” — 这对应于返回的布尔值。当你在浏览器中点击“呼叫”时，将出现以下提示：
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md:22]:
```suggestion
   首先，我们使用确认提示（confirm prompt）提示用户进行回答。这将在屏幕上显示一个窗口（如图所示），用户可以选择“确定”或“取消”——这对应于返回的布尔值。当你在浏览器中点击“呼叫”时，将出现以下提示：
```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md (hunk 1 above)
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/answer_a_call/index.md:60]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Creating_a_call", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/End_a_call")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+1. 让我们创建一个变量来存储 ID，并创建一个函数来请求用户输入它，稍后我们将调用它。将以下内容添加到 script.js 文件的底部：
+
+   ```js
+   let code;
+   function getStreamCode() {
+     code = window.prompt("Please enter the sharing code");
+   }
+   ```
+
+   window.prompt() 方法提供了一种方便的方式来获取相关的对等方 ID ——你可以在需要收集对等方 ID 以创建连接时使用它。
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md:23]:
```suggestion
   [`window.prompt()`](/zh-CN/docs/Web/API/Window/prompt) 方法提供了一种方便的方式来获取相关的对等方 ID ——你可以在需要收集对等方 ID 以创建连接时使用它。
```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+2. 使用 PeerJS 框架，你将希望将`localPeer`连接到`remotePeer`。PeerJS 为我们提供了`connect()`函数，它接受一个对等方 ID 以连接到。将以下代码块添加到上述代码的下方：
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md:25]:
```suggestion
2. 使用 PeerJS 框架，你将希望将 `localPeer` 连接到 `remotePeer`。PeerJS 为我们提供了 `connect()` 函数，它接受一个对等方 ID 以连接到。将以下代码块添加到上述代码的下方：
```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+   console.log(`Listening on: ${port}`);
+   ```
+
+5. 你应该能够通过 `localhost` 连接到你的应用程序（在我们的 `server.js` 中，我们使用端口 8000（在第 7 行定义），但你可能正在使用另一个端口号）。在终端中运行 `yarn start`（其中 `start` 是你在上一页的 `package.json` 中声明的脚本）。在浏览器中访问 `localhost:8000`，你应该会看到一个类似下面这样的页面：
+
+   ![一个奶油色的背景，标题使用深绿色粗体字显示：“phone a friend”。下方紧接着是“Connecting...”，再下方是“please use headphones!”。接着是一个大大的深绿色按钮，上面写着“Call”，与背景色相同，呈奶油色。 ](connecting_screen.png)
+
+如果你想了解更多关于 Peer.js 的信息，请查看[GitHub 上的 Peer.js 服务器存储库](https://github.com/peers/peerjs-server)。
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Setup", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.md:71]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Setup", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+3. 当创建连接时，让我们使用 PeerJS 框架的`on('connection')`来设置远程对等方的 ID 并打开连接。此侦听器的函数接受一个`connection`对象，该对象是`DataConnection`对象的实例（它是围绕 WebRTC 的[`RTCDataChannel`](/zh-CN/docs/Web/API/RTCDataChannel) 的包装器）；在此函数中，你将希望将其分配给一个变量。同样，你需要在函数之外创建变量，以便稍后进行分配。在上述代码的下方添加以下内容：
+
+   ```js
+   peer.on("connection", (connection) => {
+     conn = connection;
+   });
+   ```
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Show_hide_html", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Creating_a_call")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/create_a_peer_connection/index.md:42]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Show_hide_html", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Creating_a_call")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/creating_a_call/index.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+3. 为了测试这个功能，打开两个浏览器窗口中的 localhost:8000，并在其中一个窗口中点击“Call”。你应该会看到这个页面：
+
+   ![两个屏幕并排放置，都有浓郁的奶油色背景，标题为“phone a friend”，以黑色深绿色字体加粗显示。第一个屏幕下方显示“Your device ID is: 3b77”，第二个屏幕下方显示“Your device ID is: 2doa”，紧接着标题下方，显示“please use headphones!”。在下方，有一个大大的深绿色按钮，上面写着“Call”，字体颜色与背景色相同。第二个屏幕有一个浏览器对话框，询问对等方 ID。](screens_side_by_side.png)
+
+   如果你提交了另一个对等方的 ID，呼叫将被连接！
+
+到目前为止，一切都正常运行，但我们需要给另一个浏览器答复或拒绝呼叫。我们将在下一步中处理这个。
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Create_a_peer_connection", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Answer_a_call")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/creating_a_call/index.md:55]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Create_a_peer_connection", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Answer_a_call")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/end_a_call/index.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+   这样，如果发起通话的人首先点击了“挂断”，两个浏览器仍然会更新到新的状态。
+
+3. 再次测试你的应用程序，并尝试关闭通话。
+
+> **备注：** 在 `conn` 变量上调用的 `on('close')` 事件在 Firefox 中尚不可用；这意味着在 Firefox 中，每个呼叫者都必须单独挂断。
+
+> **警告：** 目前我们编写的代码意味着当一个连接关闭时，**只有**发起通话的人首先点击“挂断”时，两个浏览器才会更新。如果回答通话的人先点击“挂断”，则另一个呼叫者也必须点击“挂断”才能看到正确的 HTML。
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Answer_a_call", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Deployment_and_further_reading")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/end_a_call/index.md:40]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Answer_a_call", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Deployment_and_further_reading")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/get_microphone_permission/index.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    })
+    .catch((err) => {
+      console.error(`你得到了一个错误：${err}`);
+    });
+}
+
+getLocalStream();
+```
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Show_hide_html")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/get_microphone_permission/index.md:78]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Show_hide_html")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/index.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+4. 在另一个终端窗口中，通过在手机应用目录的根目录中运行以下命令来启动 Peer 服务器：
+
+   ```bash
+   peerjs --port 443 --key peerjs --path /myapp
+   ```
+
+这看起来与我们在上一步创建的 Peer 服务器非常相似；这是客户端部分。为了让浏览器连接到正在运行的 Peer 服务器，我们需要告诉它如何连接；上面的代码就是这样做的。
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Build_the_server", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Get_microphone_permission")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/index.md:44]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Build_the_server", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Get_microphone_permission")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/show_hide_html/index.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+   // 显示音频控件和正确的副本
+   function showConnectedContent() {
+     window.caststatus.textContent = "You're connected";
+     callBtn.hidden = true;
+     audioContainer.hidden = false;
+   }
+   ```
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Get_microphone_permission", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Create_a_peer_connection")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/connect_peers/show_hide_html/index.md:48]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Get_microphone_permission", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Create_a_peer_connection")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/deployment_and_further_reading/index.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+## 进一步阅读
+
+- [PeerJS](https://peerjs.com/)
+- [WebRTC](/zh-CN/docs/Web/API/WebRTC_API)
+- [PeerJS 服务器](https://github.com/peers/peerjs-server)
+- [一个类似的视频教程（带视频）](https://www.youtube.com/watch?v=OOrBcpwelPY)
+- [代码教程](https://github.com/SamsungInternet/WebPhone/tree/master/tutorial)
+
+{{PreviousMenu("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/End_a_call")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/deployment_and_further_reading/index.md:24]:
```suggestion
{{PreviousMenu("Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/End_a_call")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/setup/index.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+       "express": "^4.17.1",
+       "peer": "^0.5.3",
+       "peerjs": "^1.3.1"
+     }
+   }
+   ```
+
+4. 完成设置后，你需要将以下 [HTML](https://gist.github.com/lolaodelola/578d692e4700dfe2c9d239c80bbdbabc) 和 [CSS 文件](https://gist.github.com/lolaodelola/b4498288b86ddce995603546a64abb29)复制到项目文件夹的根目录中。你可以将这两个文件命名为 `index`，因此 HTML 文件将是 `index.html`，CSS 文件将是 `index.css`。在接下来的文章中，你不需要对这些文件进行太多修改。
+
+{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Build_the_server")}}
```
Comment (tianyeeT) [inline files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/setup/index.md:46]:
```suggestion
{{PreviousMenuNext("Web/API/WebRTC_API/Build_a_phone_with_peerjs", "Web/API/WebRTC_API/Build_a_phone_with_peerjs/Build_the_server")}}

```

File: files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/index.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+2. [连接对等体](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers)
+
+  1. [获取麦克风权限](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Get_microphone_permission)
+  2. [显示和隐藏 HTML](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Show_hide_html)
+  3. [创建对等连接](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Create_a_peer_connection)
+  4. [创建呼叫](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Creating_a_call)
+  5. [接听呼叫](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Connect_peers/Answer_a_call)
+  6. [结束呼叫](/zh-CN/docs/Web/API/WebRTC_API/build_a_phone_with_peerjs/connect_peers/End_a_call)
+
+3. [部署和进一步阅读](/zh-CN/docs/Web/API/WebRTC_API/Build_a_phone_with_peerjs/Deployment_and_further_reading)
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/webrtc_api/build_a_phone_with_peerjs/index.md]: 这里要用三个空格，我帮你修复了

**Review by yin1999 — APPROVED** [review]

**Pull request merged**